### ISSUE TITLE
Clicking on the notification should act

### ISSUE THREAD

**Issue by safwanrahman** [safwanrahman/django-webpush#29]
So currently, clicking on the notification do nothing. So it can be only used to show notification. Not possible to send a notification that user can click on the notification and they will be redirect to any link.
So we should implement that the sender can configure a url to which the receiver will be taken while he clicks on the notification.

So it will work like following:
- The sender will configure a url while sending notification. The url will be passed to the user's browser with payload.
- In the user browser it will parse the payload and get the url
- A event listener should be added to the notification
- When the user click on the event listener, he should be taken to the url as the sender provided

So it needs some backend work and most of the javascript work.

**Comment by narnikgamarnikus** [comment]
Tell me, is this not realized yet?

**Comment by safwanrahman** [comment]
@narnikgamarnikus It needs some work in javascript side and I am not that much expert in writing javascript. Can you help on this?

**Comment by narnikgamarnikus** [comment]
@safwanrahman  need to use `webpush`, and also I need to use ` "active"`. I'm very busy right now, but when I get to the `webpush`, I hope I'll do it.

**Comment by JLucasRS** [comment]
You can put this in your service worker, in the event.waitUntil.
`event.waitUntil(
    self.addEventListener('notificationclick', function (event) {
      self.clients.openWindow(data.link)
    }),
    // Show a notification with title 'ServiceWorker Cookbook' and use the payload
    // as the body.
    self.registration.showNotification(head, {
      body: body,
    })
  );
});`

**Comment by JLucasRS** [comment]
You can put this in your service worker, in the event.waitUntil:
```
event.waitUntil(
    self.addEventListener('notificationclick', function (event) {
      self.clients.openWindow(data.link)
    }),
    // Show a notification with title 'ServiceWorker Cookbook' and use the payload
    // as the body.
    self.registration.showNotification(head, {
      body: body,
    })
  );
});

**Comment by amitranjan0220** [comment]
how to add icon and icon path

**Comment by JLucasRS** [comment]
I'm not sure about what you are asking, but I managed to use icons in my notifications.
In django, when creating the payload, you can do something like this:
``` python
payload = {'head': 'Head', 'body': 'Body', 'icon': 'icon url in your server', 'url': 'url that should be open when clicked'}
```
In the service worker:
``` javascript
let url;
self.addEventListener('push', function(event) {
  let payload = event.data ? event.data.text() : {"head": "No Content", "body": "No Content", "icon": "", "url": ""},
    data = JSON.parse(payload),
    head = data.head,
    body = data.body,
    icon = data.icon; // You take the icon from the payload.

  url = data.url

  event.waitUntil(
    self.registration.showNotification(head, {
      body: body,
      icon: icon, //And then you use it the showNotification method.
    })
  );
});

self.addEventListener('notificationclick', function (event) {
  event.waitUntil(
    self.clients.openWindow(url),
    event.notification.close()
  );
})
```

In my case, I created a model to store notifications, and a ImageField to store the images called "icon". So, when creating the payload in my send_notification() method in my model class, i used icon.url to get the media url:
 ``` python
payload = {'head': self.title, 'body': self.body, 'icon': self.icon.url, 'url': self.url}

**Comment by JLucasRS** [comment]
I don't work with jinja, sorry.

**Comment by safwanrahman** [comment]
Implemented by @JLucasRS #49
